src/main/java/com/mongodb/hibernate/jdbc/MongoConnectionProvider.java

```diff
@@ -110,38 +100,53 @@ public <T> T unwrap(Class<T> unwrapType) {
     @Override
     public void configure(Map<String, Object> configValues) {
         var jdbcUrl = configValues.get(JAKARTA_JDBC_URL);
-        if (jdbcUrl == null) {
-            throw new HibernateException("Configuration is required: " + JAKARTA_JDBC_URL);
+
+        if (mongoClientCustomizer == null && jdbcUrl == null) {
+            throw new HibernateException(format(
+                    "Configuration [%s] is required unless MongoClientCustomizer is provided", JAKARTA_JDBC_URL));
```

Comment:
Instead of hardcoding the interface name in the message, let's use the `.class` reference. This way this usage of the interface is explicit and visible to the compiler.

```suggestion
                    "Configuration [%s] is required unless %s is provided",
                    JAKARTA_JDBC_URL, MongoClientCustomizer.class.getName()));
```